src/persist/src/mem.rs

```diff
@@ -582,9 +583,46 @@ impl MemMultiRegistry {
     }
 }
 
+#[derive(Debug)]
+struct MemConsensus {
+    data: Arc<Mutex<Option<VersionedData>>>,
```

Comment:
this needs to use an async friendly lock, such as https://docs.rs/tokio/latest/tokio/sync/struct.Mutex.html

Reply:
Those docs talk about how a `sync` mutex can be find if not held across an await point

In my experience, those docs are true, unless you are using the mutex as some kind of coordination point. Below it seems like we `lock` the mutex, check a seq number, and move on, which is a short critical section

Reply:
the critical section got a little bit larger with the check to make sure the sequence number is true, but its still a fairly short critical section and the lock isn't being held across `await`s so I'm going to hold off on this unless you have some specific reason Dan?

Reply:
Yes, definitely worth a comment!

Reply:
changed it to use tokio's mutex. will also change memblob to do the same in another pr